# Comparing `tmp/pciSeq-0.0.56-py3-none-any.whl.zip` & `tmp/pciSeq-0.0.57.dev0-py3-none-any.whl.zip`

## zipinfo {}

```diff
@@ -1,75 +1,75 @@
-Zip file size: 364693 bytes, number of entries: 73
--rw-rw-rw-  2.0 fat     2135 b- defN 23-Jun-25 20:54 pciSeq/__init__.py
--rw-rw-rw-  2.0 fat    10919 b- defN 23-Jul-03 14:49 pciSeq/app.py
--rw-rw-rw-  2.0 fat     3369 b- defN 23-Jul-02 23:15 pciSeq/config.py
+Zip file size: 366278 bytes, number of entries: 73
+-rw-rw-rw-  2.0 fat     2135 b- defN 23-Jul-09 15:28 pciSeq/__init__.py
+-rw-rw-rw-  2.0 fat    11165 b- defN 23-Jul-09 18:54 pciSeq/app.py
+-rw-rw-rw-  2.0 fat     3713 b- defN 23-Jul-09 19:18 pciSeq/config.py
 -rw-rw-rw-  2.0 fat     3678 b- defN 22-Apr-07 23:15 pciSeq/make_plot.py
 -rw-rw-rw-  2.0 fat        0 b- defN 22-Jan-19 17:53 pciSeq/src/__init__.py
--rw-rw-rw-  2.0 fat       22 b- defN 23-Jul-03 14:47 pciSeq/src/_version.py
+-rw-rw-rw-  2.0 fat       27 b- defN 23-Jul-09 19:27 pciSeq/src/_version.py
 -rw-rw-rw-  2.0 fat        0 b- defN 22-Jan-19 17:53 pciSeq/src/cell_call/__init__.py
--rw-rw-rw-  2.0 fat    20864 b- defN 23-Jun-29 10:21 pciSeq/src/cell_call/datatypes.py
--rw-rw-rw-  2.0 fat     2995 b- defN 23-Apr-26 22:13 pciSeq/src/cell_call/log_config.py
--rw-rw-rw-  2.0 fat    19103 b- defN 23-Jul-03 08:19 pciSeq/src/cell_call/main.py
--rw-rw-rw-  2.0 fat     3104 b- defN 23-Jun-25 20:54 pciSeq/src/cell_call/summary.py
--rw-rw-rw-  2.0 fat     9194 b- defN 23-Jun-25 20:54 pciSeq/src/cell_call/utils.py
--rw-rw-rw-  2.0 fat        0 b- defN 23-Jun-11 12:10 pciSeq/src/diagnostics/__init__.py
--rw-rw-rw-  2.0 fat      299 b- defN 23-Jun-24 12:16 pciSeq/src/diagnostics/config.py
--rw-rw-rw-  2.0 fat     3336 b- defN 23-Jun-25 20:54 pciSeq/src/diagnostics/diagnostics.py
--rw-rw-rw-  2.0 fat      796 b- defN 23-Jun-25 20:54 pciSeq/src/diagnostics/launch_diagnostics.py
--rw-rw-rw-  2.0 fat     7340 b- defN 23-Jul-03 14:46 pciSeq/src/diagnostics/utils.py
+-rw-rw-rw-  2.0 fat    22062 b- defN 23-Jul-09 18:34 pciSeq/src/cell_call/datatypes.py
+-rw-rw-rw-  2.0 fat     2995 b- defN 23-Jul-09 15:28 pciSeq/src/cell_call/log_config.py
+-rw-rw-rw-  2.0 fat    23046 b- defN 23-Jul-09 18:41 pciSeq/src/cell_call/main.py
+-rw-rw-rw-  2.0 fat     3104 b- defN 23-Jul-09 15:28 pciSeq/src/cell_call/summary.py
+-rw-rw-rw-  2.0 fat     9194 b- defN 23-Jul-09 15:28 pciSeq/src/cell_call/utils.py
+-rw-rw-rw-  2.0 fat        0 b- defN 23-Jul-09 15:28 pciSeq/src/diagnostics/__init__.py
+-rw-rw-rw-  2.0 fat      299 b- defN 23-Jul-09 15:28 pciSeq/src/diagnostics/config.py
+-rw-rw-rw-  2.0 fat     3336 b- defN 23-Jul-09 15:28 pciSeq/src/diagnostics/diagnostics.py
+-rw-rw-rw-  2.0 fat      796 b- defN 23-Jul-09 15:28 pciSeq/src/diagnostics/launch_diagnostics.py
+-rw-rw-rw-  2.0 fat     7340 b- defN 23-Jul-09 17:51 pciSeq/src/diagnostics/utils.py
 -rw-rw-rw-  2.0 fat        0 b- defN 22-Jan-19 17:53 pciSeq/src/preprocess/__init__.py
--rw-rw-rw-  2.0 fat    11253 b- defN 23-Jun-25 20:54 pciSeq/src/preprocess/cell_borders.py
+-rw-rw-rw-  2.0 fat    11253 b- defN 23-Jul-09 15:28 pciSeq/src/preprocess/cell_borders.py
 -rw-rw-rw-  2.0 fat     5323 b- defN 23-Jun-25 19:58 pciSeq/src/preprocess/regionprops.py
 -rw-rw-rw-  2.0 fat     6973 b- defN 23-Apr-26 22:13 pciSeq/src/preprocess/segmentation.py
--rw-rw-rw-  2.0 fat     4585 b- defN 23-Jun-29 10:21 pciSeq/src/preprocess/spot_labels.py
--rw-rw-rw-  2.0 fat       88 b- defN 23-Jun-29 10:21 pciSeq/src/preprocess/utils.py
+-rw-rw-rw-  2.0 fat     4585 b- defN 23-Jul-09 17:51 pciSeq/src/preprocess/spot_labels.py
+-rw-rw-rw-  2.0 fat       88 b- defN 23-Jul-09 17:51 pciSeq/src/preprocess/utils.py
 -rw-rw-rw-  2.0 fat        0 b- defN 22-Feb-20 14:12 pciSeq/src/viewer/__init__.py
--rw-rw-rw-  2.0 fat     2256 b- defN 23-Jun-25 20:54 pciSeq/src/viewer/run_flask.py
--rw-rw-rw-  2.0 fat     5007 b- defN 23-Jun-25 20:54 pciSeq/src/viewer/stage_image.py
--rw-rw-rw-  2.0 fat     9109 b- defN 23-Jul-02 20:10 pciSeq/src/viewer/utils.py
--rw-rw-rw-  2.0 fat    10895 b- defN 23-Jun-25 20:54 pciSeq/static/2D/index.html
--rw-rw-rw-  2.0 fat     6506 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/genes_datatable.html
--rw-rw-rw-  2.0 fat     9622 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/css/index.css
--rw-rw-rw-  2.0 fat     3135 b- defN 23-Jun-25 20:54 pciSeq/static/2D/viewer/css/progress.css
--rw-rw-rw-  2.0 fat     7435 b- defN 23-Jun-25 20:54 pciSeq/static/2D/viewer/js/classConfig.js
--rw-rw-rw-  2.0 fat     2314 b- defN 23-Jul-02 20:15 pciSeq/static/2D/viewer/js/config.js
--rw-rw-rw-  2.0 fat     5589 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/customControl.js
--rw-rw-rw-  2.0 fat    23501 b- defN 23-Jul-02 19:25 pciSeq/static/2D/viewer/js/dapi.js
--rw-rw-rw-  2.0 fat     5984 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/dataLoader.js
--rw-rw-rw-  2.0 fat    17232 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/donut.js
--rw-rw-rw-  2.0 fat     4106 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/dt.js
--rw-rw-rw-  2.0 fat     9835 b- defN 23-Jun-25 20:54 pciSeq/static/2D/viewer/js/glyphConfig.js
--rw-rw-rw-  2.0 fat     4389 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/glyphPaths.js
--rw-rw-rw-  2.0 fat     7224 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/glyphs.js
--rw-rw-rw-  2.0 fat    12281 b- defN 23-Jun-25 20:54 pciSeq/static/2D/viewer/js/index.js
--rw-rw-rw-  2.0 fat     1073 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/progress.js
--rw-rw-rw-  2.0 fat     6367 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/stage_cells.js
--rw-rw-rw-  2.0 fat    12064 b- defN 23-Jun-25 20:54 pciSeq/static/2D/viewer/js/stage_glyphs.js
--rw-rw-rw-  2.0 fat     9540 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/stage_markers.js
--rw-rw-rw-  2.0 fat     7580 b- defN 23-Jul-02 20:05 pciSeq/static/2D/viewer/js/stage_markers_patched.js
--rw-rw-rw-  2.0 fat    15523 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/stage_polygons.js
--rw-rw-rw-  2.0 fat     4471 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/streaming-tsv-parser.js
--rw-rw-rw-  2.0 fat     4692 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/viewerUtils.js
--rw-rw-rw-  2.0 fat      995 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/css/L.Control.Layers.Tree.css
--rw-rw-rw-  2.0 fat      421 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/css/Leaflet.Coordinates-0.1.3.css
--rw-rw-rw-  2.0 fat   122544 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/css/bootstrap.min.css
--rw-rw-rw-  2.0 fat     8669 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/css/index.css
--rw-rw-rw-  2.0 fat     1228 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/css/keen-dashboards.css
--rw-rw-rw-  2.0 fat     5245 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/css/screen.css
--rw-rw-rw-  2.0 fat    26652 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/L.Control.Layers.Tree.js
--rw-rw-rw-  2.0 fat     8813 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/Leaflet.Coordinates-0.1.3.src.js
--rw-rw-rw-  2.0 fat    31824 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/bootstrap.min.js
--rw-rw-rw-  2.0 fat    42707 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/keen.min.js
--rw-rw-rw-  2.0 fat    15374 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/leaflet-pip.js
--rw-rw-rw-  2.0 fat     5927 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/leaflet.textpath.js
--rw-rw-rw-  2.0 fat      265 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/preloader.js
--rw-rw-rw-  2.0 fat     9097 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/MarkerContainer.js
--rw-rw-rw-  2.0 fat     4434 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/bezier-easing.js
--rw-rw-rw-  2.0 fat   613808 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/example.min.js
--rw-rw-rw-  2.0 fat    91874 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/tools.min.js
--rw-rw-rw-  2.0 fat     1091 b- defN 23-Jul-03 14:50 pciSeq-0.0.56.dist-info/LICENCE
--rw-rw-rw-  2.0 fat     5274 b- defN 23-Jul-03 14:50 pciSeq-0.0.56.dist-info/METADATA
--rw-rw-rw-  2.0 fat       92 b- defN 23-Jul-03 14:50 pciSeq-0.0.56.dist-info/WHEEL
--rw-rw-rw-  2.0 fat       48 b- defN 23-Jul-03 14:50 pciSeq-0.0.56.dist-info/entry_points.txt
--rw-rw-rw-  2.0 fat        7 b- defN 23-Jul-03 14:50 pciSeq-0.0.56.dist-info/top_level.txt
-?rw-rw-r--  2.0 fat     6935 b- defN 23-Jul-03 14:50 pciSeq-0.0.56.dist-info/RECORD
-73 files, 1326430 bytes uncompressed, 353443 bytes compressed:  73.4%
+-rw-rw-rw-  2.0 fat     2256 b- defN 23-Jul-09 15:28 pciSeq/src/viewer/run_flask.py
+-rw-rw-rw-  2.0 fat     5007 b- defN 23-Jul-09 15:28 pciSeq/src/viewer/stage_image.py
+-rw-rw-rw-  2.0 fat     9109 b- defN 23-Jul-09 17:51 pciSeq/src/viewer/utils.py
+-rw-rw-rw-  2.0 fat    10895 b- defN 23-Jul-09 15:28 pciSeq/static/2D/index.html
+-rw-rw-rw-  2.0 fat     6506 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/genes_datatable.html
+-rw-rw-rw-  2.0 fat     9622 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/css/index.css
+-rw-rw-rw-  2.0 fat     3135 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/css/progress.css
+-rw-rw-rw-  2.0 fat     7716 b- defN 23-Jul-09 17:51 pciSeq/static/2D/viewer/js/classConfig.js
+-rw-rw-rw-  2.0 fat     2314 b- defN 23-Jul-09 17:51 pciSeq/static/2D/viewer/js/config.js
+-rw-rw-rw-  2.0 fat     5589 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/customControl.js
+-rw-rw-rw-  2.0 fat    23501 b- defN 23-Jul-09 17:51 pciSeq/static/2D/viewer/js/dapi.js
+-rw-rw-rw-  2.0 fat     5984 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/dataLoader.js
+-rw-rw-rw-  2.0 fat    17232 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/donut.js
+-rw-rw-rw-  2.0 fat     4106 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/dt.js
+-rw-rw-rw-  2.0 fat     9835 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/glyphConfig.js
+-rw-rw-rw-  2.0 fat     4389 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/glyphPaths.js
+-rw-rw-rw-  2.0 fat     7224 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/glyphs.js
+-rw-rw-rw-  2.0 fat    12281 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/index.js
+-rw-rw-rw-  2.0 fat     1073 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/progress.js
+-rw-rw-rw-  2.0 fat     6367 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/stage_cells.js
+-rw-rw-rw-  2.0 fat    12064 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/stage_glyphs.js
+-rw-rw-rw-  2.0 fat     9540 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/stage_markers.js
+-rw-rw-rw-  2.0 fat     7580 b- defN 23-Jul-09 17:51 pciSeq/static/2D/viewer/js/stage_markers_patched.js
+-rw-rw-rw-  2.0 fat    15523 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/stage_polygons.js
+-rw-rw-rw-  2.0 fat     4471 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/streaming-tsv-parser.js
+-rw-rw-rw-  2.0 fat     4692 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/viewerUtils.js
+-rw-rw-rw-  2.0 fat      995 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/css/L.Control.Layers.Tree.css
+-rw-rw-rw-  2.0 fat      421 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/css/Leaflet.Coordinates-0.1.3.css
+-rw-rw-rw-  2.0 fat   122544 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/css/bootstrap.min.css
+-rw-rw-rw-  2.0 fat     8669 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/css/index.css
+-rw-rw-rw-  2.0 fat     1228 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/css/keen-dashboards.css
+-rw-rw-rw-  2.0 fat     5245 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/css/screen.css
+-rw-rw-rw-  2.0 fat    26652 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/js/L.Control.Layers.Tree.js
+-rw-rw-rw-  2.0 fat     8813 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/js/Leaflet.Coordinates-0.1.3.src.js
+-rw-rw-rw-  2.0 fat    31824 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/js/bootstrap.min.js
+-rw-rw-rw-  2.0 fat    42707 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/js/keen.min.js
+-rw-rw-rw-  2.0 fat    15374 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/js/leaflet-pip.js
+-rw-rw-rw-  2.0 fat     5927 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/js/leaflet.textpath.js
+-rw-rw-rw-  2.0 fat      265 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/js/preloader.js
+-rw-rw-rw-  2.0 fat     9097 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/MarkerContainer.js
+-rw-rw-rw-  2.0 fat     4434 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/bezier-easing.js
+-rw-rw-rw-  2.0 fat   613808 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/example.min.js
+-rw-rw-rw-  2.0 fat    91874 b- defN 23-Jul-09 15:28 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/tools.min.js
+-rw-rw-rw-  2.0 fat     1091 b- defN 23-Jul-09 19:36 pciSeq-0.0.57.dev0.dist-info/LICENCE
+-rw-rw-rw-  2.0 fat     5279 b- defN 23-Jul-09 19:36 pciSeq-0.0.57.dev0.dist-info/METADATA
+-rw-rw-rw-  2.0 fat       92 b- defN 23-Jul-09 19:36 pciSeq-0.0.57.dev0.dist-info/WHEEL
+-rw-rw-rw-  2.0 fat       48 b- defN 23-Jul-09 19:36 pciSeq-0.0.57.dev0.dist-info/entry_points.txt
+-rw-rw-rw-  2.0 fat        7 b- defN 23-Jul-09 19:36 pciSeq-0.0.57.dev0.dist-info/top_level.txt
+?rw-rw-r--  2.0 fat     6965 b- defN 23-Jul-09 19:36 pciSeq-0.0.57.dev0.dist-info/RECORD
+73 files, 1332482 bytes uncompressed, 354968 bytes compressed:  73.4%
```

## zipnote {}

```diff
@@ -195,26 +195,26 @@
 
 Filename: pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/example.min.js
 Comment: 
 
 Filename: pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/tools.min.js
 Comment: 
 
-Filename: pciSeq-0.0.56.dist-info/LICENCE
+Filename: pciSeq-0.0.57.dev0.dist-info/LICENCE
 Comment: 
 
-Filename: pciSeq-0.0.56.dist-info/METADATA
+Filename: pciSeq-0.0.57.dev0.dist-info/METADATA
 Comment: 
 
-Filename: pciSeq-0.0.56.dist-info/WHEEL
+Filename: pciSeq-0.0.57.dev0.dist-info/WHEEL
 Comment: 
 
-Filename: pciSeq-0.0.56.dist-info/entry_points.txt
+Filename: pciSeq-0.0.57.dev0.dist-info/entry_points.txt
 Comment: 
 
-Filename: pciSeq-0.0.56.dist-info/top_level.txt
+Filename: pciSeq-0.0.57.dev0.dist-info/top_level.txt
 Comment: 
 
-Filename: pciSeq-0.0.56.dist-info/RECORD
+Filename: pciSeq-0.0.57.dev0.dist-info/RECORD
 Comment: 
 
 Zip file comment:
```

## pciSeq/app.py

```diff
@@ -262,18 +262,25 @@
 
 if __name__ == "__main__":
     # set up the logger
     attach_to_log()
 
     # read some demo data
     _iss_spots = pd.read_csv(os.path.join(ROOT_DIR, 'data', 'mouse', 'ca1', 'iss', 'spots.csv'))
-    _coo = load_npz(os.path.join(ROOT_DIR, 'data', 'mouse', 'ca1', 'segmentation', 'label_image.coo.npz'))
+    spots = pd.read_csv(r"F:\data\Sara\20230707\spotsT007.csv")
 
-    _scRNAseq = pd.read_csv(os.path.join(ROOT_DIR, 'data', 'mouse', 'ca1', 'scRNA', 'scRNAseq.csv.gz'),
-                            header=None, index_col=0, compression='gzip', dtype=object)
-    _scRNAseq = _scRNAseq.rename(columns=_scRNAseq.iloc[0], copy=False).iloc[1:]
-    _scRNAseq = _scRNAseq.astype(float).astype(np.uint32)
+    dat = np.load(r'F:\data\Sara\20230707\background_image_fixed_seg 2.npy', allow_pickle=True).item()
+    masks = dat['masks']
+    coo = coo_matrix(masks)
+    # _coo = load_npz(os.path.join(ROOT_DIR, 'data', 'mouse', 'ca1', 'segmentation', 'label_image.coo.npz'))
+    #
+    # _scRNAseq = pd.read_csv(os.path.join(ROOT_DIR, 'data', 'mouse', 'ca1', 'scRNA', 'scRNAseq.csv.gz'),
+    #                         header=None, index_col=0, compression='gzip', dtype=object)
+    # _scRNAseq = _scRNAseq.rename(columns=_scRNAseq.iloc[0], copy=False).iloc[1:]
+    # _scRNAseq = _scRNAseq.astype(float).astype(np.uint32)
 
     # main task
     # _opts = {'max_iter': 10}
-    fit(spots=_iss_spots, coo=_coo, scRNAseq=_scRNAseq, opts={'save_data': True,'launch_viewer': True,})
+    fit(spots=spots, coo=coo, opts={'save_data': True,'launch_viewer': True, 'launch_diagnostics': True})
+
+
```

## pciSeq/config.py

```diff
@@ -60,9 +60,17 @@
     # Use either np.float16 or np.float32 to reduce memory usage. In most cases RAM consumption shouldnt
     # need more than 32Gb RAM. If you have a dataset from a full coronal mouse slice with a high number of
     # segmented cells (around 150,000) a gene panel of more than 250 genes and 100 or more different
     # cell types (aka clusters, aka classes) in the single cell data then you might need at least 64GB on
     # your machine. Changing the datatype to a float16 or float32 will help keeping RAM usage to a lower
     # level
     'dtype': np.float64,
+
+
+    # *******************************************************************************
+    # Hyperparameters below come into action **ONLY** if single cell data are missing
+    # *******************************************************************************
+    'mean_gene_counts_per_class': 60,
+    'mean_gene_counts_per_cell': 15,
+
 }
```

## pciSeq/src/_version.py

```diff
@@ -1 +1 @@
-__version__ = '0.0.56'
+__version__ = '0.0.57.dev0'
```

## pciSeq/src/cell_call/datatypes.py

```diff
@@ -1,26 +1,28 @@
 import sys
 import numpy as np
 import pandas as pd
 import scipy
 import numexpr as ne
+import numpy_groupies as npg
 from natsort import natsort_keygen
 from sklearn.neighbors import NearestNeighbors
 from pciSeq.src.cell_call.log_config import logger
 
 
 class Cells(object):
     # Get rid of the properties where not necessary!!
     def __init__(self, _cells_df, config):
         self.config = config
         self.ini_cell_props, self.mcr = self.read_image_objects(_cells_df, config)
         self.nC = len(self.ini_cell_props['cell_label'])
         self.classProb = None
         self.class_names = None
         self._cov = self.ini_cov()
+        self.nu_0 = config['mean_gene_counts_per_cell']
         self._centroid = self.ini_centroids()
         self._gene_counts = None
         self._background_counts = None
 
     # -------- PROPERTIES -------- #
     @property
     def yx_coords(self):
@@ -80,28 +82,47 @@
 
     def nn(self):
         n = self.config['nNeighbors'] + 1
         # for each spot find the closest cell (in fact the top nN-closest cells...)
         nbrs = NearestNeighbors(n_neighbors=n, algorithm='ball_tree').fit(self.yx_coords)
         return nbrs
 
-    # def geneCountsPerKlass(self, single_cell_data, egamma, ini):
-    #     # ********************************************
-    #     # DEPRECATED. Replaced by a simple einsum call
-    #     # ********************************************
-    #     temp = np.einsum('ck, c, cgk -> gk', self.classProb, self.cell_props['area_factor'], egamma)
-    #
-    #     # total counts predicted by all cells of each class (nG, nK)
-    #     ClassTotPredicted = temp * (single_cell_data.mean_expression + ini['SpotReg'])
-    #
-    #     # total of each gene
-    #     isZero = ClassTotPredicted.columns == 'Zero'
-    #     labels = ClassTotPredicted.columns.values[~isZero]
-    #     TotPredicted = ClassTotPredicted[labels].sum(axis=1)
-    #     return TotPredicted
+    def scatter_matrix(self, spots):
+        mu_bar = self.centroid.values
+        prob = spots.parent_cell_prob[:, :-1]
+        id = spots.parent_cell_id[:, :-1]
+        xy_spots = spots.xy_coords
+        out = self.ini_cov() * self.nu_0
+
+        mu_x = mu_bar[id, 0]  # array of size [nS, N] with the x-coord of the centroid of the N closest cells
+        mu_y = mu_bar[id, 1]  # array of size [nS, N] with the y-coord of the centroid of the N closest cells
+
+        N = mu_x.shape[1]
+        _x = np.tile(xy_spots[:, 0], (N, 1)).T  # array of size [nS, N] populated with the x-coord of the spot
+        _y = np.tile(xy_spots[:, 1], (N, 1)).T  # array of size [nS, N] populated with the y-coord of the spot
+        x_centered = _x - mu_x  # subtract the cell centroid x-coord from the spot x-coord
+        y_centered = _y - mu_y  # subtract the cell centroid y-coord from the spot x-coord
+
+        el_00 = prob * x_centered * x_centered  # contribution to the scatter matrix's [0, 0] element
+        el_11 = prob * y_centered * y_centered  # contribution to the scatter matrix's off-diagonal element
+        el_01 = prob * x_centered * y_centered  # contribution to the scatter matrix's [1, 1] element
+
+        # Aggregate all contributions to get the scatter matrix
+        agg_00 = npg.aggregate(id.ravel(), el_00.ravel(), size=self.nC)
+        agg_11 = npg.aggregate(id.ravel(), el_11.ravel(), size=self.nC)
+        agg_01 = npg.aggregate(id.ravel(), el_01.ravel(), size=self.nC)
+
+        # Return now the scatter matrix. Some cell might not have any spots nearby. For those empty cells,
+        # the scatter matrix will be a squared zero array. That is fine.
+        out[:, 0, 0] = agg_00
+        out[:, 1, 1] = agg_11
+        out[:, 0, 1] = agg_01
+        out[:, 1, 0] = agg_01
+
+        return out
 
     def read_image_objects(self, img_obj, cfg):
         meanCellRadius = np.mean(np.sqrt(img_obj.area / np.pi)) * 0.5
         relCellRadius = np.sqrt(img_obj.area / np.pi) / meanCellRadius
 
         # append 1 for the misreads
         relCellRadius = np.append(1, relCellRadius)
@@ -477,15 +498,15 @@
         return out
 
     def _diag(self, genes):
         # logger.info('******************************************************')
         # logger.info('*************** DIAGONAL SINGLE CELL DATA ***************')
         # logger.info('******************************************************')
         nG = len(genes)
-        mgc = 15  # the avg gene count per cell. Better expose that so it can be set by the user.
+        mgc = self.config['mean_gene_counts_per_class']  # the avg gene count per cell. Better expose that so it can be set by the user.
         arr = mgc * np.eye(nG)
         labels = ['class_%d' % (i+1) for i, _ in enumerate(genes)]
         df = pd.DataFrame(arr).set_index(genes)
         df.columns = labels
         return df
 
 # ---------------------------------------- Class: CellType --------------------------------------------------- #
```

## pciSeq/src/cell_call/main.py

```diff
@@ -62,14 +62,17 @@
 
             # 1. For each cell, calc the expected gene counts
             self.geneCount_upd()
 
             # 2. calc expected gamma
             self.gamma_upd()
 
+            if self.single_cell.isMissing:
+                self.gaussian_upd()
+
             # 3. assign cells to cell types
             self.cell_to_cellType()
 
             if self.single_cell.isMissing:
                 self.dalpha_upd()
 
             # 4. assign spots to cells
@@ -250,14 +253,106 @@
         # res = num / denom
 
         # Finally, update gene_gamma
         self.genes.calc_eta(alpha, beta)
         # self.genes.eta_bar = res.values
 
     # -------------------------------------------------------------------- #
+    def gaussian_upd(self):
+        self.centroid_upd()
+        self.cov_upd()
+
+    # -------------------------------------------------------------------- #
+    def centroid_upd(self):
+        spots = self.spots
+
+        # get the total gene counts per cell
+        N_c = self.cells.total_counts
+
+        xy_spots = spots.xy_coords
+        prob = spots.parent_cell_prob
+        n = self.cells.config['nNeighbors'] + 1
+
+        # multiply the x coord of the spots by the cell prob
+        a = np.tile(xy_spots[:, 0], (n, 1)).T * prob
+
+        # multiply the y coord of the spots by the cell prob
+        b = np.tile(xy_spots[:, 1], (n, 1)).T * prob
+
+        # aggregated x and y coordinate
+        idx = spots.parent_cell_id
+        x_agg = npg.aggregate(idx.ravel(), a.ravel(), size=len(N_c))
+        y_agg = npg.aggregate(idx.ravel(), b.ravel(), size=len(N_c))
+
+        # x_agg = np.zeros(N_c.shape)
+        # mask = np.arange(len(_x_agg))
+        # x_agg[mask] = _x_agg
+        #
+        # y_agg = np.zeros(N_c.shape)
+        # mask = np.arange(len(_y_agg))
+        # y_agg[mask] = _y_agg
+
+        # get the estimated cell centers
+        x_bar = np.nan * np.ones(N_c.shape)
+        y_bar = np.nan * np.ones(N_c.shape)
+
+        x_bar[N_c > 0] = x_agg[N_c > 0] / N_c[N_c > 0]
+        y_bar[N_c > 0] = y_agg[N_c > 0] / N_c[N_c > 0]
+
+        # cells with N_c = 0 will end up with x_bar = y_bar = np.nan
+        xy_bar_fitted = np.array(list(zip(x_bar.T, y_bar.T)))
+
+        # if you have a value for the estimated centroid use that, otherwise
+        # use the initial (starting values) centroids
+        ini_cent = self.cells.ini_centroids()
+        xy_bar = np.array(tuple(zip(*[ini_cent['x'], ini_cent['y']])))
+
+        # # sanity check. NaNs or Infs should appear together
+        # assert np.all(np.isfinite(x_bar) == np.isfinite(y_bar))
+        # use the fitted centroids where possible otherwise use the initial ones
+        xy_bar[np.isfinite(x_bar)] = xy_bar_fitted[np.isfinite(x_bar)]
+        self.cells.centroid = pd.DataFrame(xy_bar, columns=['x', 'y'])
+        # print(np.array(list(zip(x_bar.T, y_bar.T))))
+
+    # -------------------------------------------------------------------- #
+    def cov_upd(self):
+        spots = self.spots
+
+        # first get the scatter matrix
+        S = self.cells.scatter_matrix(spots)  # sample sum of squares
+        cov_0 = self.cells.ini_cov()
+        nu_0 = self.cells.nu_0
+        S_0 = cov_0 * nu_0  # prior sum of squares
+        N_c = self.cells.total_counts
+        d = 2
+        denom = np.ones([self.nC, 1, 1])
+        denom[:, 0, 0] = N_c + nu_0
+        # Note: need to add code to handle the case N_c + nu_0 <= d + 2
+        cov = (S + S_0) / denom
+
+        # delta = self.cells.ledoit_wolf(self.spots, cov)
+        # stein = self.cells.stein(cov)
+        # delta = self.cells.rblw(cov)
+
+        delta = 0.5
+        # logger.info('Mean shrinkage %4.2f' % delta.mean())
+        # logger.info('cell 601 shrinkage %4.2f, %4.2f' % (shrinkage[601], sh[601]))
+        #  logger.info('cell 601 gene counts %d' % self.cells.total_counts[601])
+        # logger.info('cell 605 shrinkage %4.2f, %4.2f' % (shrinkage[605], sh[605]))
+        #  logger.info('cell 605 gene counts %d' % self.cells.total_counts[605])
+        # logger.info('cell 610 shrinkage %4.2f, %4.2f' % (shrinkage[610], sh[610]))
+        #  logger.info('cell 610 gene counts %d' % self.cells.total_counts[610])
+
+        # delta = delta.reshape(self.nC, 1, 1)
+        # target = [np.trace(d)/2 * np.eye(2) for d in cov]  # shrinkage target
+        cov = delta * cov_0 + (1 - delta) * cov
+        # cov = delta * target + (1 - delta) * cov
+        self.cells.cov = cov
+
+    # -------------------------------------------------------------------- #
     def mu_upd(self):
         # logger.info('Update single cell data')
         # # make an array nS-by-nN and fill it with the spots id
         # gene_ids = np.tile(self.spots.gene_id, (self.nN, 1)).T
         #
         # # flatten it
         # gene_ids = gene_ids.ravel()
@@ -302,25 +397,25 @@
         alpha = self.cellTypes.ini_alpha()
         out = zeta + alpha
 
         # logger.info("***************************************************")
         # logger.info("**** Dirichlet alpha is zero if class size <=%d ****" % self.config['min_class_size'])
         # logger.info("***************************************************")
 
-        # 07-May-2013: Hiding 'min_class_size' from the config file. Should bring it back at a later version
+        # 07-May-2023: Hiding 'min_class_size' from the config file. Should bring it back at a later version
         # mask = zeta <= self.config['min_class_size']
         min_class_size = 5
         mask = zeta <= min_class_size
 
         # make sure Zero class is the last one
         assert self.cellTypes.names[-1] == "Zero"
         assert len(self.cellTypes.names) == len(mask)
 
         # make sure the last value ie always False, overriding if necessary the
-        # check a few lines when the mask variable was set.
+        # check a few lines above when the mask variable was set.
         # In this manner we will prevent the Zero class from being removed.
         mask[-1] = False
 
         # If a class size is smaller than 'min_class_size' then it will be assigned a weight of almost zero
         out[mask] = 10e-6
         self.cellTypes.alpha = out
```

## pciSeq/static/2D/viewer/js/classConfig.js

### js-beautify {}

```diff
@@ -1,10 +1,13 @@
 // Color scheme for the cell classes.
 // IdentifiedType is some wider categorization. On the viewer, the donut chart at the bottom right is aggregated
 // by IdentifiedType and these aggregated probabilities are shown on the table at the bottom left.
+// Also, on the viewer, at the top right, there is a collapsible control to switch on/off the cells by the
+// cell class. The control has the classes in a nested class/subClass manner. For the nesting operation to
+// work, the separator in the className string should be a dot.
 
 function classColorsCodes() {
     var out = [
 
         {
             className: 'Pvalb.C1ql1.Cpne5',
             IdentifiedType: 'Axo-axonic',
```

## Comparing `pciSeq-0.0.56.dist-info/LICENCE` & `pciSeq-0.0.57.dev0.dist-info/LICENCE`

 * *Files identical despite different names*

## Comparing `pciSeq-0.0.56.dist-info/METADATA` & `pciSeq-0.0.57.dev0.dist-info/METADATA`

 * *Files 2% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 2.1
 Name: pciSeq
-Version: 0.0.56
+Version: 0.0.57.dev0
 Summary: Probabilistic cell typing for spatial transcriptomics
 Home-page: https://github.com/acycliq/pciSeq
 Author: Dimitris Nicoloutsopoulos
 Author-email: dimitris.nicoloutsopoulos@gmail.com
 License: BSD
 Classifier: Programming Language :: Python :: 3
 Classifier: License :: OSI Approved :: BSD License
@@ -94,15 +94,15 @@
 `memurai.exe â€“service-start`
 
 If redis (or memurai) is missing from your system, the call to launch the diagnostics dashboard will be 
 ignored. If you are interested in this feature you may find this notebook [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/acycliq/pciSeq/blob/master/notebooks/4_diagnostics.ipynb)
  useful
 
 ## Change Log
-### [0.0.55] - 2023-07-03
+### [0.0.56] - 2023-07-03
  - Diagnostics dashboard
 
  - Baselayers on the viewer. You can have multiple background images and switch between them.
 
 ### [0.0.50] - 2023-05-27
  - Single cell data are optional, more info can be found here [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/acycliq/pciSeq/blob/master/notebooks/3_pciSeq_without_singleCell_data.ipynb)
```

## Comparing `pciSeq-0.0.56.dist-info/RECORD` & `pciSeq-0.0.57.dev0.dist-info/RECORD`

 * *Files 7% similar despite different names*

```diff
@@ -1,17 +1,17 @@
 pciSeq/__init__.py,sha256=mxqjE8smLqDQuwzheJ_iA3gfcyV7c4x8lHQPwSEQwJk,2135
-pciSeq/app.py,sha256=OsFFq_u6K-G3SNWW7m4wX0P4p2fqhTOyMqT0cqKEy_I,10919
-pciSeq/config.py,sha256=u227HnbaSSGJs3wPopJju7GK59WPl7lN6G27JTkoPZc,3369
+pciSeq/app.py,sha256=CaJQSw3_D8HOxy1yIFBvTYaW9hhObAtW0mDxsoqyIdY,11165
+pciSeq/config.py,sha256=UgT_7SEMMImfuMTIpCw3_t-DYMRMtbZ-SqBjq0rsGMQ,3713
 pciSeq/make_plot.py,sha256=2AkncrVVvytr2keK505oMC0LV1Txli9WpsSGpjmht_4,3678
 pciSeq/src/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
-pciSeq/src/_version.py,sha256=Fe52qmmbSgJdsNu8U2KzWwnm9t8Jah4ESzC4LpmSaLc,22
+pciSeq/src/_version.py,sha256=D0ZlQn6xfQdpbfv9NDewCTqsljokvldKNO7o1C7xkk4,27
 pciSeq/src/cell_call/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
-pciSeq/src/cell_call/datatypes.py,sha256=om8kMxOUxKcAoQHSTgXEuX4UNMGtwTjFkOCGmeublqs,20864
+pciSeq/src/cell_call/datatypes.py,sha256=tpTSWdEVL_JozrX9kx23AniU5TcMb-GFRBRg2HnOHNw,22062
 pciSeq/src/cell_call/log_config.py,sha256=vnm9TxlMEVG_GLPfl21GDEWTafEk3B-drTLsVQMfFEE,2995
-pciSeq/src/cell_call/main.py,sha256=PgOk1dngmqk357AYCZxmGFoFjyZBgW_nEKNb5F0CKK4,19103
+pciSeq/src/cell_call/main.py,sha256=0IA8rtIkeEiQ1nGfuN5vcxlayx1HcdAqdI_t5KodK7c,23046
 pciSeq/src/cell_call/summary.py,sha256=f2snLqJIeywtYuHrJLM-hRvifgqRU2u2aCKE_JrZJPs,3104
 pciSeq/src/cell_call/utils.py,sha256=M4KJrS82pIHuCoxg2ej6NE9fJAVCvJw90T7YmWI2T7o,9194
 pciSeq/src/diagnostics/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
 pciSeq/src/diagnostics/config.py,sha256=kfPzvlFdGN0QgdfECwVV8lujXUiyim4fuetJoU4ziBQ,299
 pciSeq/src/diagnostics/diagnostics.py,sha256=L8tBgykN-KktrUS8YdIP5doi8meVVUUZU2PvOcUt72g,3336
 pciSeq/src/diagnostics/launch_diagnostics.py,sha256=p2t1wDc48iWvpfEjUt_-J5iK37EXPMg2O12y-pCeE_w,796
 pciSeq/src/diagnostics/utils.py,sha256=fNolyKf8SDl-W113NRxvEEwm9J5rxK_K_-a35YigI68,7340
@@ -25,15 +25,15 @@
 pciSeq/src/viewer/run_flask.py,sha256=9HS_RW-O9xddAeiApd4m_w2JJmbqUGZ5lhQ-z_HUeVA,2256
 pciSeq/src/viewer/stage_image.py,sha256=0zvdVk_R52T43ueSt3K3DgdMZPUyJNxDSyG0Bni7O78,5007
 pciSeq/src/viewer/utils.py,sha256=VwHLGIVQJnaRYeDAjoIoP87O8qXkeS7H4IHJynQPjsw,9109
 pciSeq/static/2D/index.html,sha256=9aMR5hOaXugJpSOt3ypUY24z6blsfOS83Cd9LHbAHa0,10895
 pciSeq/static/2D/viewer/genes_datatable.html,sha256=v10lTjkXYeAv0DVuaWHOB6XgG887k-_ABXAgobDV-tY,6506
 pciSeq/static/2D/viewer/css/index.css,sha256=IntuFc7kNAaNkLSL3m_9fpnH67i5KJpNWJ0OIW4TmD4,9622
 pciSeq/static/2D/viewer/css/progress.css,sha256=0MfjX9kIwD4wBeC39LBi9bCvu4ZkHwg_vtbuUiy4CuI,3135
-pciSeq/static/2D/viewer/js/classConfig.js,sha256=xhfWJHcU1bXipgJEokkMdUsrPDifPHpp_UHzW2qYopg,7435
+pciSeq/static/2D/viewer/js/classConfig.js,sha256=QuGSikifUK9McS_rBVCqwsoCiUl-89mncmlyYauYMMo,7716
 pciSeq/static/2D/viewer/js/config.js,sha256=mhuRBtP_lLeEQzSWYb9D28C4dfufLu3KLFu2Edr0Zik,2314
 pciSeq/static/2D/viewer/js/customControl.js,sha256=niQd9FT0z_Dkp7v523n8plj7kNRnRpvADcHOChbDskY,5589
 pciSeq/static/2D/viewer/js/dapi.js,sha256=m3y5z6GU2VQ7H74HKhDq_9RH0xNivt_TaZWM-HcNPDY,23501
 pciSeq/static/2D/viewer/js/dataLoader.js,sha256=iD3Kv9XNsSD2YyU3ilau5iaxAEQEX5VJnQ06VWyEP-g,5984
 pciSeq/static/2D/viewer/js/donut.js,sha256=dn9S3ZnNTFdEaiDU7ZadoKjVyvYp9LEJm82EvCe1ySY,17232
 pciSeq/static/2D/viewer/js/dt.js,sha256=eJpyKHNJjs1jf-GiTEa2_UsxMadO5eRWs2J0sJMhFms,4106
 pciSeq/static/2D/viewer/js/glyphConfig.js,sha256=23YX0DGc7XQXuJrUJDt59lref_TZPwUGF_Pr4h8c-tk,9835
@@ -61,13 +61,13 @@
 pciSeq/static/2D/viewer/js/lib/js/leaflet-pip.js,sha256=QqsilU0SHTaKA_qNSv_jEDH53cWO6T_RdforfybnbKM,15374
 pciSeq/static/2D/viewer/js/lib/js/leaflet.textpath.js,sha256=i5cmMPqpqXShZz0TvXU3ZQUi2Y07SIbIpMYbhPE8gWE,5927
 pciSeq/static/2D/viewer/js/lib/js/preloader.js,sha256=eaFWRf9yT-Td4C2t01P8fDEZYszYTaGDgQ1vjevNIDI,265
 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/MarkerContainer.js,sha256=Z8TADCqYEQr25OEF7wljsOVAOpx87UrjFPI2nvIty70,9097
 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/bezier-easing.js,sha256=yMoRDqJiglg0M0eWM25WGA5zzLWuJpOyC3a8aGFx0jw,4434
 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/example.min.js,sha256=bZEnK8XSnShrpLYtOS7Xq7wnp-nW_nMZh4aaDc1Gt6U,613808
 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/tools.min.js,sha256=A027Ub-oWXvKbz2B3pgMFS5xApmnEHXz9GQh79YZZnU,91874
-pciSeq-0.0.56.dist-info/LICENCE,sha256=6kbiFSfobTZ7beWiKnHpN902HgBx-Jzgcme0SvKqhKY,1091
-pciSeq-0.0.56.dist-info/METADATA,sha256=aKRhNbSLiQszynA8KTo7Slmrfb-BGPmF1DmN-O5ADeo,5274
-pciSeq-0.0.56.dist-info/WHEEL,sha256=G16H4A3IeoQmnOrYV4ueZGKSjhipXx8zc8nu9FGlvMA,92
-pciSeq-0.0.56.dist-info/entry_points.txt,sha256=RGOkAd0S7rrTuEInj-jKu5B7630i4R3mK9v9vx6nqc8,48
-pciSeq-0.0.56.dist-info/top_level.txt,sha256=mSonMN0SaB51m5yLGm5hFE1Fthf-2U4sRKxSezY-JMk,7
-pciSeq-0.0.56.dist-info/RECORD,,
+pciSeq-0.0.57.dev0.dist-info/LICENCE,sha256=6kbiFSfobTZ7beWiKnHpN902HgBx-Jzgcme0SvKqhKY,1091
+pciSeq-0.0.57.dev0.dist-info/METADATA,sha256=VxMViJO-Wjq9ryYkbdR9qSWasZOuHAtp2DuVeFyAhZM,5279
+pciSeq-0.0.57.dev0.dist-info/WHEEL,sha256=G16H4A3IeoQmnOrYV4ueZGKSjhipXx8zc8nu9FGlvMA,92
+pciSeq-0.0.57.dev0.dist-info/entry_points.txt,sha256=RGOkAd0S7rrTuEInj-jKu5B7630i4R3mK9v9vx6nqc8,48
+pciSeq-0.0.57.dev0.dist-info/top_level.txt,sha256=mSonMN0SaB51m5yLGm5hFE1Fthf-2U4sRKxSezY-JMk,7
+pciSeq-0.0.57.dev0.dist-info/RECORD,,
```

